pymc/distributions/discrete.py

```diff
@@ -1370,23 +1377,28 @@ def logp(value, c):
             -np.inf,
         )
 
-
-class ZeroInflatedPoissonRV(RandomVariable):
-    name = "zero_inflated_poisson"
-    ndim_supp = 0
-    ndims_params = [0, 0]
-    dtype = "int64"
-    _print_name = ("ZeroInflatedPois", "\\operatorname{ZeroInflatedPois}")
-
-    @classmethod
-    def rng_fn(cls, rng, psi, lam, size):
-        return rng.poisson(lam, size=size) * (rng.random(size=size) < psi)
+    def logcdf(value, c):
+        return at.switch(
+            at.lt(value, c),
+            -np.inf,
+            0,
+        )
 
 
-zero_inflated_poisson = ZeroInflatedPoissonRV()
+def _zero_inflated_mixture(*, name, nonzero_p, nonzero_dist, **kwargs):
+    nonzero_p = at.as_tensor_variable(floatX(nonzero_p))
+    weights = at.stack([1 - nonzero_p, nonzero_p], axis=-1)
+    comp_dists = [
+        Constant.dist(0),
+        nonzero_dist,
```

Comment:
I think your question meet be related to https://github.com/pymc-devs/pymc/issues/4511

Still need to make sure we are not moving discrete and continuous distributions in mixtures. In the cases here they are both discrete yes

Reply:
Did you have another thing in mind?

Reply:
This is exactly what I had in mind! I was wondering if it would help to have a check that `nonzero_dist` is discrete, but, as you mentioned above, it may not be necessary since `_zero_inflated_mixture` is not a user-facing function.